src/System.Collections.Specialized/src/System/Collections/Specialized/BitVector32.cs

```diff
@@ -174,7 +171,7 @@ private static Section CreateSectionHelper(short maxValue, short priorMask, shor
 #if DEBUG
             int maskCheck = CreateMaskFromHighValue(maxValue);
             int offsetCheck = priorOffset + CountBitsSet(priorMask);
-            Debug.Assert(maskCheck <= short.MaxValue && offsetCheck < 32, "Overflow on BitVector32");
+            Debug.Assert(maskCheck <= short.MaxValue, "Overflow on BitVector32");
```

Comment:
It is covered in the lines following it - this throws a InvalidOperationException if I recall correctly

Thus this was an improper assert and resulted in different behaviour in release and debug